deepstream/5.1/deepstream_ssd_parser.py

```diff
@@ -0,0 +1,518 @@
+    f.close()
+    return labels
+
+
+def make_elm_or_print_err(factoryname, name, printedname, detail=""):
+    """ Creates an element with Gst Element Factory make.
+        Return the element  if successfully created, otherwise print
+        to stderr and return None.
+    """
+    print("Creating", printedname)
+    elm = Gst.ElementFactory.make(factoryname, name)
+    if not elm:
+        sys.stderr.write("Unable to create " + printedname + " \n")
+        if detail:
+            sys.stderr.write(detail)
+    return elm
+
+
+def osd_sink_pad_buffer_probe(pad, info, u_data):
+    frame_number = 0
+    # Intiallizing object counter with 0.
+    obj_counter = dict(enumerate([0] * CLASS_NB))
+    num_rects = 0
+
+    gst_buffer = info.get_buffer()
+    if not gst_buffer:
+        print("Unable to get GstBuffer ")
+        return
+
+    # Retrieve batch metadata from the gst_buffer
+    # Note that pyds.gst_buffer_get_nvds_batch_meta() expects the
+    # C address of gst_buffer as input, which is obtained with hash(gst_buffer)
+    batch_meta = pyds.gst_buffer_get_nvds_batch_meta(hash(gst_buffer))
+    l_frame = batch_meta.frame_meta_list
+    while l_frame is not None:
+        try:
+            # Note that l_frame.data needs a cast to pyds.NvDsFrameMeta
+            # The casting also keeps ownership of the underlying memory
+            # in the C code, so the Python garbage collector will leave
+            # it alone.
+            frame_meta = pyds.NvDsFrameMeta.cast(l_frame.data)
+        except StopIteration:
+            break
+
+        frame_number = frame_meta.frame_num
+        num_rects = frame_meta.num_obj_meta
+        l_obj = frame_meta.obj_meta_list
+        while l_obj is not None:
+            try:
+                # Casting l_obj.data to pyds.NvDsObjectMeta
+                obj_meta = pyds.NvDsObjectMeta.cast(l_obj.data)
+            except StopIteration:
+                break
+            obj_counter[obj_meta.class_id] += 1
+            try:
+                l_obj = l_obj.next
+            except StopIteration:
+                break
+
+        # Acquiring a display meta object. The memory ownership remains in
+        # the C code so downstream plugins can still access it. Otherwise
+        # the garbage collector will claim it when this probe function exits.
+        display_meta = pyds.nvds_acquire_display_meta_from_pool(batch_meta)
+        display_meta.num_labels = 1
+        py_nvosd_text_params = display_meta.text_params[0]
+        # Setting display text to be shown on screen
+        # Note that the pyds module allocates a buffer for the string, and the
+        # memory will not be claimed by the garbage collector.
+        # Reading the display_text field here will return the C address of the
+        # allocated string. Use pyds.get_string() to get the string content.
+        id_dict = {
+            val: index
+            for index, val in enumerate(get_label_names_from_file("ssd_mobilenet_v2/labels.txt"))
```

Comment:
Please don't use hard coded path. Put the input argument instead even if this is a constant address.

Reply:
done.